### ISSUE TITLE
Font Size

### ISSUE THREAD

**Issue by sagecedar** [davidsansome/tsurukame#334]
I’ve been using your app for quite a while and loved it.  I increased the font size to largest size but didn’t see any change.  I’m low vision so font size is important to me.  Anyway, just this morning, it was like, WHOA!  It’s giant!  But, the rest is still not correct.   I mean, if I made a typo, the info it displayed - the definition and kana are still too small (or remain unchanged.)
![Uploading 5C99884E-A9E1-4723-92CE-79C93D50EF62.png…]
![352265BD-14FA-4829-8BF2-8F4762A9C899](https://user-images.githubusercontent.com/25251300/77855771-83076b00-71a7-11ea-900d-b9af08ca234e.png). I use iPad Pro, by the way.

**Comment by turtlemaster19** [comment]
@sagecedar Yep, that's the font size. We can tell it's an iPad Pro, but I agree the font size setting could be better done. Leaving this open to comment.

**Comment by sagecedar** [comment]
Just wanted to clarify that while font size works great for the main kanji. What I would like to see is ability to change fonts for the rest, such as the meaning and kana, etc.  Does that make sense?

Thanks,

Jeremy

**Comment by turtlemaster19** [comment]
@sagecedar Definitely, that's why I don't want to close it, since that screenshot looks awful on the iPad with that font size.

**Comment by jeff8v7** [comment]
Totally agree.  On an iPad pro's large screen, the text is illegibly small at times.  I can read the English, but cannot make out the complex Kanji, especially with the low contrast red on dark grey motif.

For size comparison, look at the clock and "< Back" button in the upper left.  These are the smallest fonts generally used on iPad – and only in this simple case for controls which should not get your attention.  

The main font for this app is smaller than the "< Back" button.  For iPad pro, roughly double would be considered normal.

![E70CC7EB-8D58-4E8A-88EA-D50D29303EE3](https://user-images.githubusercontent.com/39231776/79082716-97c92e80-7d63-11ea-9f8a-725de4e23827.jpeg)

For comparison... 
![359C8572-0D5F-4C54-9516-87B9AAF807D3](https://user-images.githubusercontent.com/39231776/79082719-a283c380-7d63-11ea-96dd-1489c6b228aa.jpeg)

**Comment by jeff8v7** [comment]
By the way... another font size slider probably isn't needed, as most apps don't have one.  Just make the default text size relative to the screen size...

PS: I know the blurred out text is intentionally illegible in the first screen shot.  I wasn't referring to that :-)

**Comment by jeff8v7** [comment]
@sagecedar  I noticed that the text is seems more legible with the "light" theme.  This might be a quick improvement for you :-)

![B7FAFC5D-4A9C-4928-B0CF-63E503F58217](https://user-images.githubusercontent.com/39231776/79083186-97329700-7d67-11ea-8a39-a0ce9c566485.png)

**Comment by turtlemaster19** [comment]
@jeff8v7 I don't think that makes much of a difference. :P

**Comment by jeff8v7** [comment]
Really? My eyes must be pretty bad..

**Comment by turtlemaster19** [comment]
@jeff8v7 I just meant switching the theme, sorry. Maybe I'll look at my own iPad's lessons without my glasses and see...

**Comment by turtlemaster19** [comment]
@jeff8v7 I finally saw what you meant when I tried it on my phone without glasses today. I couldn't say it was super noticeable on the iPad, because the screen size was so large, that the effect was diminished.

Those are some good points you made, and I hope that we can fix this soon...

**Comment by aaron-meyers** [comment]
Bumping this up - I use Tsurukame on an iPad mini and the text is tiny. I made the text size bigger in the iPad settings but Tsurukame doesn't seem to respect this as it has no effect (I've noticed it changes the text size in some but not all apps). I don't really care whether Tsurukame respects the iOS text settings or has its own setting but would be really great to support larger text throughout the app.

**Comment by jeff8v7** [comment]
The font size setting seems to apply to the item being tested only. On a large iPad, the item is actually uncomfortably large, while all other text is uncomfortable small.

**Comment by geerk** [comment]
Is it possible to utilize Dynamic Font features? So that any text can be dynamically scaled using built-in a11y features of the iOS? I found some related article: https://mobilea11y.com/guides/swiftui/swiftui-dynamic-type/
I could give it a try with the support from the authors (because I have no experience in iOS development, only web-dev).

**Comment by UInt2048** [comment]
@geerk We do some of that already.

**Comment by geerk** [comment]
@UInt2048 I mean in the screenshot from the issue description some labels use dynamic font so they have larger font (probably because they do not have font size specified [here](https://github.com/davidsansome/tsurukame/blob/1b44e43fdd0dfb1ab94d6e4a61e979d7ebc59cb0/ios/SubjectDetailsView.swift#L167)), but the text content actually stays unchanged probably because the font size is hard-coded [here](https://github.com/davidsansome/tsurukame/blob/1b44e43fdd0dfb1ab94d6e4a61e979d7ebc59cb0/ios/SubjectDetailsView.swift#L20).
